### PR TITLE
libfreerdp/crypto/certificate.c: handle malloc error

### PR DESCRIPTION
**Pull request by chipitsine**

### CODE REVIEW

File: libfreerdp/crypto/certificate.c [hunk 1]
```diff
@@ -1615,6 +1615,11 @@ BOOL freerdp_certificate_publickey_encrypt(const rdpCertificate* cert, const BYT
 
 	size_t outputSize = EVP_PKEY_size(pkey);
 	output = malloc(outputSize);
+	if (output == NULL)
+	{
+		EVP_PKEY_CTX_free(ctx);
```
Comment (akallabeth) [inline libfreerdp/crypto/certificate.c:1620]: use `goto out;` instead.

**Review by akallabeth — CHANGES_REQUESTED** [review]
Please group your changes in one pull request if you have more of these small fixes in queue.

**Comment by chipitsine** [comment]
> Please group your changes in one pull request if you have more of these small fixes in queue.

this was the last one

File: libfreerdp/crypto/certificate.c (hunk 1 above)
Comment (chipitsine) [reply libfreerdp/crypto/certificate.c:1620]: makes sense. give me 5 min

**Comment by hardening** [comment]
Looks ok for me but the history must be cleaned with just 1 commit with the 1 line change

**Comment by akallabeth** [comment]
@freerdp-bot test

**Review by hardening — APPROVED** [review]

**Review by akallabeth — APPROVED** [review]

**Pull request merged**